### PR TITLE
[US133639] AQD: Table sorting

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US133639](https://rally1.rallydev.com/#/?detail=/userstory/614624045065&fdp=true): AQD: Table sorting

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
    * [x] Screen readers announce sorting order
    * [x] It is possible to change sorting with keyboard space or enter keys
  * [x] RTL
  * [x] Responsive
  * [x] Test cases
    * [x] Each table appears sorted by the first column ascending by default (the same as the User table in LMS)
    * [x] When a user sorts a column the first page is set
    * [x] Sort icon appears before tooltip icon
    * [x] Clicking on a column header sorts the column ascending
    * [x] A click on a sorted column toggles the sort order 
* Security, devops, perf: N/A
* UX and docs

* [x] Demo

  
FYI @Brightspace/draco-vis

### AQ vs ED vs Users

It seems we need to change EQ behaviour

image

### CODE REVIEW

**Comment by Vitalii-Misechko** [comment]
Working on failing tests

File: pages/assessment-quality-dashboard-view.js [hunk 1]
```diff
@@ -57,7 +58,16 @@ const tableColumnFiltersMap = [
...
+export const TABLE_COLUMNS = {
+	QUIZ_NAME: 'dashboard:quizDetailsQuiz',
+	AVG_SCORE: 'dashboard:quizDetailsAverageScore',
+	RELIABILITY_COEF: 'dashboard:quizDetailsReliabilityCoefficient',
+	QUIZ_ITEMS_NUM: 'dashboard:quizDetailsNumberOfQuizItems',
+	// FLAGGED_ITEMS: 'dashboard:quizDetailsFlaggedQuizItems',
+	LEARNER_ANSWERS_NUM: 'dashboard:quizDetailsNumberOfLearnerAnswers'
+};
+
+class AssessmentQualityDashboardView extends SortMixin(SkeletonMixin(Localizer(MobxLitElement))) {
```
Comment (NicholasHallman) [inline pages/assessment-quality-dashboard-view.js:70]: Glad the sort mixin was useable here

**Review by johngwilkinson — APPROVED** [review]

File: components/table/assessment-quiz-table.js [hunk 2]
```diff
@@ -116,35 +157,95 @@ class AssessmentTable extends SkeletonMixin(Localizer(MobxLitElement)) {
...
 		`;
 	}
 
-	_renderHeaderWithToolTip(uniqueId, headerLangTerm, tooltipTextLangTerm) {
-		const tooltipId = `d2l-table-tooltip-${uniqueId}`;
+	/**
+	 * @param {TableHeaderDesc} header
+	 * @param {Number} idx
+	 */
+	_renderHeaderNoToolTip(header, idx) {
```
Comment (bemailloux) [inline components/table/assessment-quiz-table.js:176]: Looks like there is a lot of shared code between `_renderHeaderNoToolTip` and `_renderHeaderWithToolTip`

**Review by bemailloux — APPROVED** [review]

File: components/table/assessment-quiz-table.js (hunk 2 above)
Comment (Vitalii-Misechko) [reply components/table/assessment-quiz-table.js:176]: You're right. These two share some amount of common code and can be expressed in one function. But my inner threshold of commonality hasn't triggered yet ) I find these two more readable when they are separate.

File: pages/assessment-quality-dashboard-view.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply pages/assessment-quality-dashboard-view.js:70]: > Glad the sort mixin was useable here

yeah ... I liked that idea with mixin. So, I borrowed it from ED

**Pull request merged**